Update MatrixChainProblem.java

sknoll-eng opened the pull request.

sknoll-eng committed:
Update MatrixChainProblem.java

actions-user committed:
Auto-format code with Spotless

offqueue commented:
This doesn't compile. There is an undefined variable "k" at line 432. How do you test the code? I fixed that one variable on my machine, but the program is unable to display a matrix chain problem.
In LCSProblem, there is only one enum of execution states, regardless of whether you are working forward or if you are backtracking. I think that is probably important because you can check the execution state at any point by looking at one variable "executionState". The way you've done it, there are two execution states that you would have to keep track of to determine where you are in the execution, which seems like it would muddy the waters. Unless there is a good reason for it that I can't see because I can't run the code.

sknoll-eng committed:
Update MatrixChainProblem.java

Fix undefined k and merge backtracking into main execution state enum. Ensures MatrixChain compiles and runs like the other problems.

actions-user committed:
Auto-format code with Spotless

sknoll-eng commented:
@offqueue Thanks for catching that. The k reference was my mistake from a refactor and I missed it before pushing. I will fix that and recompile to make sure the problem loads in the tutor. Your point on the execution states makes sense. Keeping everything in one state enum is cleaner and matches how the other problems work. I will collapse the backtracking states into the main enum and push the update.
